[mdoc] 5.7 has an issue with NullReferenceException

Running the latest release of `mdoc` on .NET platform assemblies results in a `System.NullReferenceException`.

```
2018-06-17T21:56:48.6786674Z Updating: System.Collections.Immutable.IImmutableStack`1
2018-06-17T21:56:48.6786674Z Updating: System.Collections.Immutable.ImmutableArray
2018-06-17T21:56:48.7099179Z Updating: System.Collections.Immutable.ImmutableArray`1
2018-06-17T21:56:49.3973963Z ##[error]mdoc: System.NullReferenceException: Object reference not set to an instance of an object.
   at Mono.Documentation.Updater.MemberFormatter._AppendTypeName(StringBuilder buf, TypeReference type, DynamicParserContext context, Boolean appendGeneric)
   at Mono.Documentation.Updater.MemberFormatter.GetTypeName(TypeReference type, DynamicParserContext context, Boolean appendGeneric)
   at Mono.Documentation.Updater.MemberFormatter.GetMethodDeclaration(MethodDefinition method)
   at Mono.Documentation.Updater.CSharpFullMemberFormatter.GetMethodDeclaration(MethodDefinition method)
   at Mono.Documentation.MDocUpdater.<>c__DisplayClass95_0.<DoUpdateType2>b__4(MemberReference m)
   at System.Linq.Enumerable.WhereEnumerableIterator`1.MoveNext()
   at System.Linq.Buffer`1..ctor(IEnumerable`1 source)
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at Mono.Documentation.MDocUpdater.DoUpdateType2(String message, XmlDocument basefile, TypeDefinition type, FrameworkTypeEntry typeEntry, String output, Boolean insertSince)
   at Mono.Documentation.MDocUpdater.DoUpdateType(TypeDefinition type, FrameworkTypeEntry typeEntry, String basepath, String dest)
   at Mono.Documentation.MDocUpdater.DoUpdateAssembly(AssemblySet assemblySet, AssemblyDefinition assembly, XmlElement index_types, String source, String dest, HashSet`1 goodfiles)
   at Mono.Documentation.MDocUpdater.DoUpdateAssemblies(String source, String dest)
   at Mono.Documentation.MDocUpdater.Run(IEnumerable`1 args)
   at Mono.Documentation.MDoc.Main(String[] args)
See `mdoc help' for more information or use --debug to diagnose.
```

--- a/mdoc/Consts.cs
+++ b/mdoc/Consts.cs
@@ -3,7 +3,7 @@ namespace Mono.Documentation
 {
 	public static class Consts
 	{
-		public static string MonoVersion = "5.7";
+		public static string MonoVersion = "5.7.0.1";
 		public const string DocId = "DocId";
 		public const string CppCli = "C++ CLI";
 	    public const string CppCx = "C++ CX";

--- a/mdoc/Mono.Documentation/Updater/Formatters/FSharpFormatter.cs
+++ b/mdoc/Mono.Documentation/Updater/Formatters/FSharpFormatter.cs
@@ -228,9 +228,11 @@ namespace Mono.Documentation.Updater
 
             foreach (var interfaceImplementation in type.Interfaces)
             {
+                var resolvedInterface = interfaceImplementation.InterfaceType.Resolve ();
+
                 if (type.IsValueType
                     && ignoredValueTypeInterfaces.Any(i => interfaceImplementation.InterfaceType.FullName.StartsWith(i))
-                    || interfaceImplementation.InterfaceType.Resolve().IsNotPublic)
+                    || (resolvedInterface != null && resolvedInterface.IsNotPublic))
                     continue;
                 buf.Append($"{GetLineEnding()}{Consts.Tab}interface ");
                 AppendTypeName(buf, GetTypeName(interfaceImplementation.InterfaceType));

--- a/mdoc/Mono.Documentation/Updater/Formatters/MemberFormatter.cs
+++ b/mdoc/Mono.Documentation/Updater/Formatters/MemberFormatter.cs
@@ -106,6 +106,9 @@ namespace Mono.Documentation.Updater
 
         protected StringBuilder _AppendTypeName (StringBuilder buf, TypeReference type, DynamicParserContext context, bool appendGeneric = true)
         {
+            if (type == null)
+                return buf;
+            
             if (type is ArrayType)
             {
                 return AppendArrayTypeName(buf, type, context);
@@ -129,7 +132,9 @@ namespace Mono.Documentation.Updater
             {
                 try
                 {
-                    type = type.Resolve();
+                    var rtype = type.Resolve ();
+                    if (rtype != null)
+                        type = rtype;
                 }
                 catch (Exception)
                 {
@@ -137,6 +142,7 @@ namespace Mono.Documentation.Updater
                     // It seems, they never have `type.IsRequiredModifier == true`, but just in case.
                 }
             }
+
 
             if (type.GenericParameters.Count == 0 &&
                     (genInst == null ? true : genInst.GenericArguments.Count == 0))

--- a/mdoc/mdoc.nuspec
+++ b/mdoc/mdoc.nuspec
@@ -2,7 +2,7 @@
 <package >
   <metadata>
     <id>mdoc</id>
-    <version>5.7</version>
+    <version>5.7.0.1</version>
     <title>mdoc</title>
     <authors>Microsoft</authors>
     <owners>Microsoft</owners>

